src/components/learner-credit-management/invite-modal/InviteMembersModalWrapper.jsx

```diff
@@ -0,0 +1,124 @@
+import React, { useCallback, useContext, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ActionRow, Button, FullscreenModal, Hyperlink, StatefulButton, useToggle,
+} from '@edx/paragon';
+import { getConfig } from '@edx/frontend-platform/config';
+import { snakeCaseObject } from '@edx/frontend-platform/utils';
+
+import { useBudgetId, useSubsidyAccessPolicy } from '../data';
+import InviteModalContent from './InviteModalContent';
+import SystemErrorAlertModal from '../cards/assignment-allocation-status-modals/SystemErrorAlertModal';
+import LmsApiService from '../../../data/services/LmsApiService';
+import { BudgetDetailPageContext } from '../BudgetDetailPageWrapper';
+
+const InviteMembersModalWrapper = ({ isOpen, close }) => {
+  const { subsidyAccessPolicyId } = useBudgetId();
+  const { data: subsidyAccessPolicy } = useSubsidyAccessPolicy(subsidyAccessPolicyId);
+  const [learnerEmails, setLearnerEmails] = useState([]);
+  const [canInviteMembers, setCanInviteMembers] = useState(false);
+  const [inviteButtonState, setInviteButtonState] = useState('default');
+  const [isSystemErrorModalOpen, openSystemErrorModal, closeSystemErrorModal] = useToggle(false);
+  const {
+    successfulInvitationToast: { displayToastForInvitation },
+  } = useContext(BudgetDetailPageContext);
+
+  const handleCloseInviteModal = () => {
+    close();
+    setInviteButtonState('default');
+  };
+
+  // Callback function for when emails are changed in the
+  // child AssignmentModalContent component. Must be memoized as
+  // the function is used within a `useEffect`'s dependency array.
+  const handleEmailAddressesChanged = useCallback((
+    value,
+    { canInvite = false } = {},
+  ) => {
+    setLearnerEmails(value);
+    setCanInviteMembers(canInvite);
+  }, []);
+
+  const handleInviteMembers = async () => {
+    setInviteButtonState('pending');
+
+    const payload = snakeCaseObject({
+      learnerEmails: learnerEmails.join(','),
+    });
+    const formData = new FormData();
+    Object.keys(payload).forEach(key => formData.append(key, payload[key]));
```

Comment:
Can use [snakeCaseFormData](https://github.com/openedx/frontend-app-admin-portal/blob/135cb6ec381b0a36db7b19d214030aaa4dfba4b3/src/utils.js#L180) for this

Reply:
like so: 
```suggestion
    const formData = snakeCaseFormData({
      learnerEmails: learnerEmails.join(','),
    });
```

Reply:
nit: Per discussions offline, it looks like [snakeCaseFormData](https://github.com/openedx/frontend-app-admin-portal/blob/135cb6ec381b0a36db7b19d214030aaa4dfba4b3/src/utils.js#L180) only supports FormData, not plain objects, so in this case I'd create a helper method for converting plain objects to FormData, for something along the lines of
```suggestion
    const formData = toFormData(snakeCaseObject({
      learnerEmails: learnerEmails.join(','),
    }));
```